plugins/arElasticSearchPlugin/lib/model/arElasticSearchInformationObject.class.php

```diff
@@ -36,12 +38,30 @@ public function load()
     $this->count = QubitPdo::fetchColumn($sql, array(QubitInformationObject::ROOT_ID));
   }
 
+  static function loadTerms($options = array())
+  {
+    $sql  = 'SELECT term.id, term.parent_id';
+    $sql .= ' FROM '.QubitTerm::TABLE_NAME;
+
+    if (isset($options['taxonomyIds']) && is_array($options['taxonomyIds']))
+    {
+      $sql .= ' WHERE '.QubitTerm::TABLE_NAME.'.taxonomy_id IN ('.implode(',',$options['taxonomyIds']).')';
+    }
+    $rows = QubitPdo::fetchAll($sql, array() ,array('fetchMode' => PDO::FETCH_ASSOC));
+    foreach ($rows as $row)
+    {
+      $terms[$row['id']] = $row['parent_id'];
+    }
+
+    return $terms;
+  }
+
   public function populate()
   {
     $this->load();
-
+    $this->terms = self::loadTerms(array('taxonomyIds' => array(QubitTaxonomy::SUBJECT_ID, QubitTaxonomy::PLACE_ID, QubitTaxonomy::GENRE_ID)));
     // Recursively descend down hierarchy
-    $this->recursivelyAddInformationObjects(QubitInformationObject::ROOT_ID, $this->count);
+    $this->recursivelyAddInformationObjects(QubitInformationObject::ROOT_ID, $this->count, array('terms' => $this->terms));
```

Comment:
The value of option['terms'] always seems to be equal to `$this->terms`, so I think it's better to call `$this->terms` directly inside `recursivelyAddInformationObjects()` and not pass the term list as a parameter.

Reply:
I have made this change for the populate() method, but not for the update() method as it is static.